clients/cloud/rust/src/consumer.rs

```diff
@@ -0,0 +1,57 @@
+extern crate futures;
+extern crate rdkafka;
+extern crate tokio;
+
+use futures::stream::Stream;
+use rdkafka::consumer::stream_consumer::StreamConsumer;
+use rdkafka::consumer::Consumer;
+use rdkafka::Message;
+use std::boxed::Box;
+use tokio::runtime::current_thread::Runtime;
+
+mod utils;
+
+fn echo_message<M: Message>(msg: M) -> Result<(), std::str::Utf8Error> {
+    let deserialize = |o| match o {
+        None => Ok("".to_string()),
+        Some(val) => Ok(std::str::from_utf8(val)?.to_string()),
```

Comment:
If I do that, then I need to use `.to_vec()` which then means I need add type annotation to the closure. Hence, it's less code to have it as is.

Reply:
Hmm on second thought, we're taking a `&[u8]` here, converting it to a `&str` and then to a `String`. I think we should be able to avoid the `to_string()` copying the `&str` to a `String` here right?

Reply:
ah good observation - no need to convert to a `String`